src/test/java/com/flightstats/hub/util/IntegrationUdpServer.java

```diff
@@ -1,73 +1,66 @@
 package com.flightstats.hub.util;
 
 import lombok.Builder;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import lombok.extern.slf4j.Slf4j;
 
 import java.io.BufferedReader;
 import java.io.ByteArrayInputStream;
 import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.net.DatagramPacket;
 import java.net.DatagramSocket;
 import java.io.IOException;
-import java.util.concurrent.TimeUnit;
-import java.util.stream.Collectors;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
 
+@Slf4j
 @Builder
 public class IntegrationUdpServer {
-    private final static Logger logger = LoggerFactory.getLogger(IntegrationUdpServer.class);
-    private long timeoutMillis;
-    private final List<String> queue = new ArrayList<>();
     private int port;
-    private boolean listening;
 
     private final Map<String, String> store = new HashMap<>();
 
-    private final CompletableFuture<Map<String, String>> serverFuture = CompletableFuture.supplyAsync(() -> {
-        try {
-            DatagramSocket serverSocket = new DatagramSocket(port);
-            while (listening) {
-                byte[] data = new byte[70];
-                String result = openSocket(data, serverSocket);
-                Thread.sleep(500);
-                addValueToStore(result);
+    public CompletableFuture<Map<String, String>> getServerFuture(CountDownLatch startupCountDownLatch, ExecutorService executorService) {
+        return  CompletableFuture.supplyAsync(() -> {
+            boolean listening = true;
+            try {
+                DatagramSocket serverSocket = new DatagramSocket(port);
+                while (listening) {
+                    byte[] data = new byte[70];
+                    DatagramPacket receivePacket = new DatagramPacket(data, data.length);
+                    startupCountDownLatch.countDown();
+                    serverSocket.receive(receivePacket);
+
+                    String result = listen(receivePacket);
+                    addValueToStore(result);
+
+                    if (result.contains("closeSocket")) {
+                        listening = false;
```

Comment:
Sort of nit picky, but if this is the only way to exit the loop, you can remove listening completely and just ```break``` here.